JS-week-2/candy_helper.js

```diff
@@ -0,0 +1,43 @@
+let boughtCandy = [];
+
+function addCandy(candyType, weight){
+    let cadies = ['sweet','chocolate','toffee','chewing_gum'];
+    let candyPricePerGram = [0.5, 0.7, 0.3, 0.03];
+
+   // let candyIndex = cadies.findIndex((element) => element === candyType);
+   let candyIndex;
+for (let index = 0; index < cadies.length; index++) {
+    if(cadies[index] === candyType){
+        candyIndex = index;
+        break;
+    }
+
+}
+
+    boughtCandy.push(candyPricePerGram[candyIndex] * weight);
```

Comment:
If you decide to use the `for` approach, it's important to handle cases when `candyType` isn't valid. Let's see what happens if we call `addCandy("onion", 10)`:


```js
let candyIndex; // it starts with `undefined`
for (let index = 0; index < cadies.length; index++) {
    if(cadies[index] === candyType){
        candyIndex = index; // if candy type isn't in the array, this line will never run
        break;
    }
}

// so here: 
//   `candyIndex` is `undefined`
//   `candyPricePerGram[candyIndex]` is also `undefined`
//   `candyPricePerGram[candyIndex] * weight` is `NaN` 
boughtCandy.push(candyPricePerGram[candyIndex] * weight);
```